cpp/tests/test_abm_world.cpp

```diff
@@ -403,6 +403,95 @@ TEST(TestWorld, evolveMigration)
         EXPECT_EQ(p_severe.get_infection_state(t + dt), mio::abm::InfectionState::Dead);
         EXPECT_EQ(p_severe.get_location().get_type(), mio::abm::LocationType::Cemetery);
     }
+
+    // Test when the NPIs are applied, people cannot enter targeted location
+    {
```

Comment:
This test is already huge, can we maybe not add more to it? Please make this its own test.

Reply:
I move it to a new test at the end of the file.

Reply:
The test name kind of sounds like this belongs into test_location, do you think it would be better to be more specific? That is, the test checks whether evolve (specifically migrate) correctly deals with compliance, correct? You could base the name off of that.

Reply:
I changed the name and added some test cases for the Trip(s) as well.